plugins/wporg-5ftf/includes/pledge-meta.php

```diff
@@ -516,9 +523,16 @@ function enqueue_assets() {
 		'before'
 	);
 
-	$current_page = get_current_screen();
-	if ( Pledge\CPT_ID === $current_page->id ) {
-		wp_enqueue_style( '5ftf-admin' );
-		wp_enqueue_script( '5ftf-admin' );
+	if ( is_admin() ) {
+		$current_page = get_current_screen();
+		if ( Pledge\CPT_ID === $current_page->id ) {
+			wp_enqueue_style( '5ftf-admin' );
+			wp_enqueue_script( '5ftf-admin' );
+		}
+	} else {
+		global $post;
+		if ( $post instanceof WP_Post && has_shortcode( $post->post_content, '5ftf_pledge_form_manage' ) ) {
+			wp_enqueue_script( '5ftf-admin' );
```

Comment:
I'm not sure if this is actually a security issue or not, but if someone went to the `/manage-pledge` URL without actually being logged in or having a valid auth token, they would still get this enqueued script, which has inline data added that includes the `manageNonce` value. Would it work to also add a `can_manage_pledge` check here before enqueuing?

Reply:
It shouldn't matter, because even if they make the nonce'd request to an endpoint, we check for `can_manage_pledge` before continuing to do any action. But fewer scripts to server is always better, so I'll add the check here too.